### ISSUE TITLE
kafka: missing/changed attributes

### ISSUE THREAD

**Issue by Dariusch** [aiven/terraform-provider-aiven#879]
## What happened?

So I provisioned a aiven_kafka instance via terraform.
My instance was initially created. But for some reason my port and access_cert changed somewhere along the way.

I cannot access Kafka on the port I was given in `aiven_kafka.kafka.service_port`.
The correct port can be accessed via a transformation `{ for i in data.aiven_kafka.kafka.components : i.component => i }.kafka.port` (which is not pretty).

Also, I do not have access to the access_cert via terraform. The certificate in the UI (XHR `service.connection_info.kafka_access_cert`) differs from the one I get via the data source `data.aiven_kafka.kafka.kafka.0.access_cert`.

## What did you expect to happen?

1. The updated port being accessible in e.g `data.aiven_kafka.kafka.service_port` or via a map (not a list)
2. The updated certificate (accessible in the UI) being available via the data source

## What else do we need to know?

Basically, I try to populate a GCP Secret Manager Secret with the Aiven Kafka values to be picked up via the [external-secrets-operator](https://github.com/external-secrets/external-secrets/) and populated into my k8s cluster.

I authenticate with Kafka via SASL, I have no public access but VPC Peering enabled.
I use the aiven provider in version 3.4.0.

**Comment by Dariusch** [comment]
Well...
Basically I found the solution in the deep dive of the [aiven docs](https://docs.aiven.io/docs/products/kafka/howto/kafka-sasl-auth.html).. it seems like you need to enable certificate auth with SASL, without it won't work.
Additionallty the correct ca.crt to be used is not on the aiven_kafka but on the [aiven_project](https://registry.terraform.io/providers/aiven/aiven/latest/docs/data-sources/project#ca_cert) data/resource.
The ones I was referring to are the ones to be used with the [Client Certificate](https://registry.terraform.io/providers/aiven/aiven/latest/docs/data-sources/kafka#access_cert) authentication and not with SASL.

To get the SASL port just use the following: `data.aiven_kafka.kafka.components[1].port` (hope you never change the order)
The above suggested solution won't work with SASL and Client Certificate enabled, as the component is returning duplicate maps with `component = "kafka"`
